language/vm/vm-genesis/src/lib.rs

```diff
@@ -258,35 +258,22 @@ pub fn encode_genesis_transaction_with_validator(
             txn_data.sender = genesis_addr;
 
             let mut txn_executor = TransactionExecutor::new(&gas_schedule, &data_cache, txn_data);
```

Comment:
I would try to get rid of the transaction executor in the future as genesis is the only code that uses such functionality right now.

Reply:
yes we need to write the genesis generation in a way that is more "tuned" with the runtime